worker-markup-core/src/main/java/com/github/cafdataprocessing/worker/markup/core/MarkupHeadersAndBody.java

```diff
@@ -137,15 +144,27 @@ private void markUpHeadersAndBody(final Element parentElement)
 
             // Check to see if the a field name is surrounded with '*' and it's value is split onto two lines,
             // each line containing an email address.
-            final Matcher splitFromFieldMatcher = fromFieldSplitOntoTwoLines.matcher(emailText);
+            if (splitFromFieldMatcher == null) {
+                splitFromFieldMatcher = fromFieldSplitOntoTwoLines.matcher(emailText);
+            } else {
+                splitFromFieldMatcher = splitFromFieldMatcher.reset(emailText);
+            }
 
-            if (splitFromFieldMatcher.find()) {
-                handleHeaderWithAsterisks(emailText, lines, fromHeaderFieldValues);
+            if (emailText.contains(GENERAL_CHECK_ONE) && emailText.contains(GENERAL_CHECK_TWO) && emailText.contains(GENERAL_CHECK_THREE)) {
```

Comment:
Does this condition really improve the performance?  It looks like it would have to make multiple passes over the text content to make these checks.

Reply:
It is true, but the pattern contained in splitFromFieldMatcher is very complex, and I think it may be worthwhile to use contains (even three times) instead of find. Let me know if you want me to remove it.

Reply:
Leave it if you think it improves the performance (presumably it must be causing the regular expression to not be queried at all in many cases if it is doing so).